src/rules/valid-expect.ts

```diff
@@ -362,6 +361,25 @@ export default createRule<[Options], MessageIds>({
           // if we didn't warn user already
           !promiseArrayExceptionExists(finalNode.loc)
         ) {
+          const functionExpression = findFirstFunctionExpression(finalNode);
+
+          if (functionExpression && !functionExpression.async) {
+            context.report({
+              loc: functionExpression.loc,
+              data: { orReturned },
+              messageId:
+                finalNode === targetNode
+                  ? 'asyncMustBeAwaited'
+                  : 'promisesWithAsyncAssertionsMustBeAwaited',
+              node,
+              fix(fixer) {
+                const targetFunction =
+                  getNormalizeFunctionExpression(functionExpression);
+
+                return fixer.insertTextBefore(targetFunction, 'async ');
+              },
+            });
+          }
```

Comment:
@G-Rath 
I've attempted to modify the logic, but is my understanding correct?

Reply:
No this isn't correct, since we're now reporting an error for the function which is 1. not the actual issue and 2. reported per expect (which itself already reports an error).

Reply:
I see. Do I need to add both `async` and `await` within a single `context.report`?

Reply:
Yes

Reply:
Thanks.
>  if I add the async to the function, then both awaits get added correctly.

Is my understanding of adding async in the following code correct?

```ts
fixer.insertTextBefore(targetFunction, 'async ');
```

Reply:
Yup that looks like it's probably right, though it's not the only way.

Reply:
> my guess is that it's because you're adding the async in the first fixer, which ends up changing the logic in the second fixer - if I add the async to the function, then both awaits get added correctly.

Could you please confirm my understanding? Are you pointing out the order of the fixer's array to return? Is adding async first, as shown below, the issue?

```ts
const fixes: RuleFix[] = [];

if (!functionExpression.async) {
  const targetFunction =
    getNormalizeFunctionExpression(functionExpression);

  fixes.push(fixer.insertTextBefore(targetFunction, 'async '));
}
return [...fixes, fixer.insertTextBefore(finalNode, 'await ')];
 ```

Reply:
I don't know what the issue is, otherwise I'd have already fixed this 😅

You'll either need to dig into it or leave this with me to dig into when I next have the bandwidth.

If you're finding it overwhelming, keep in mind you can technically "do what you want" locally - sometimes I delete all the tests for a rule or create a brand new one where I can focus on a particular thing and explore what ESLint is doing with console log etc.

https://astexplorer.net/ is also very useful - I often use the "transform" option to debug particular fixers:

![image](https://github.com/jest-community/eslint-plugin-jest/assets/3151613/c160dad6-a14a-4134-8095-2a46f13f7089)

Also make sure to have a look at other rules to see how they're doing things and just consider stuff from another angle - in this case, since you're having trouble fixing when there are multiple expects, you might decide to focus on autofixing only when there is one expect; that might turn out to be harder to detect within the `CallExpression` since you can't "unreport" an error, but in that case you could push the results into a global array and then do the reporting in `Program:exit` (that's something a few of our other rules do) - it's not ideal but if it works its an improvement.